torchtune/training/_grad_scaler.py

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from torch import nn
+
+
+def scale_grads(model: nn.Module, scaler: torch.Tensor) -> None:
+    """
+    Utility to scale the gradients of a model.
+    This is useful for gradient accumulation where we want to normalize
+    the gradients by the total number of tokens seen.
+
+    Inputs:
+        model (nn.Module): model whose gradients should be scaled
+        scaler (torch.Tensor): scaling factor to apply to the gradients
+
+    Outputs:
+        None (grad fields are modified in place)
+    """
+    for p in model.parameters():
+        if p.grad is not None:
+            p.grad *= scaler
```

Comment:
Is there any concern here around overflows for lower dtypes? We could do a scaler range check based on dtype. Or  is it better to leave it to the recipe to safely choose scaler values?

Reply:
Discussed a bit offline, but this should be safe for bf16 and fp32. We can revisit upon integration of other dtypes